### PR TITLE
feat: add dashboard auto refresh behind feature flag

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #3693 

### Description:

Adds feature flag check `dashboard-auto-refresh`
Adds client-only auto-refresh functionality - if the user refreshes the page, the auto-refresh settings are reset and back to `Manual`.
Once an interval is created, it will run: 
```ts
clearCacheAndFetch();
invalidateDashboardRelatedQueries();
```

Screen-recording

Refresh-every-1-minute 

https://github.com/lightdash/lightdash/assets/7611706/e2f49caf-e68e-44d9-8964-b3f96d96ab4c

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/Dashboard/DashboardRefreshButton.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+    const isOneAtLeastFetching = isFetching > 0;
+
+    const invalidateAndSetRefreshTime = useCallback(() => {
+        clearCacheAndFetch();
+        invalidateDashboardRelatedQueries();
+        setLastRefreshTime(new Date());
+    }, [clearCacheAndFetch, invalidateDashboardRelatedQueries]);
+
+    useEffect(() => {
```
Comment (ZeRego) [inline packages/frontend/src/components/common/Dashboard/DashboardRefreshButton.tsx:62]: Would the mantine useInterval hook work here? https://v6.mantine.dev/hooks/use-interval/

Comment (joaoviana) [reply packages/frontend/src/components/common/Dashboard/DashboardRefreshButton.tsx:62]: ooohh I like that! Let me check 😉

Comment (joaoviana) [reply packages/frontend/src/components/common/Dashboard/DashboardRefreshButton.tsx]: seems to work well!

**Comment by joaoviana** [comment]
need to remove some dead code

**Review by ZeRego — APPROVED** [review]

**Pull request merged**